sklearn/gaussian_process/kernels.py

```diff
@@ -1686,8 +1686,9 @@ def __call__(self, X, Y=None, eval_gradient=False):
                 D = squareform(dists**2)[:, :, np.newaxis]
 
             if self.nu == 0.5:
-                K_gradient = K[..., np.newaxis] * D \
-                    / np.sqrt(D.sum(2))[:, :, np.newaxis]
+                denominator = np.sqrt(D.sum(2))[:, :, np.newaxis]
+                K_gradient = K[..., np.newaxis] * \
+                    np.divide(D, denominator, where=denominator != 0)
                 K_gradient[~np.isfinite(K_gradient)] = 0
```

Comment:
This fix allows us to get rid of this line
```suggestion
```

Reply:
@jeremiedbb 

Thanks for your suggestion.
I applied it.